pkg/store/kubeconfig_store_capi.go

```diff
@@ -128,9 +137,18 @@ func (s *CapiStore) StartSearch(channel chan storetypes.SearchResult) {
 	clusters := &clusterv1beta1.ClusterList{}
 	err := s.Client.List(ctx, clusters)
 	if err != nil {
+		// if kubeconfigPath for mgmt cluster is defined error out and return if the Cluster CRD is not installed
+		// otherwise silently fail as our current context might not have CAPI installed
+		if s.Config.KubeconfigPath != "" {
+			channel <- storetypes.SearchResult{
+				KubeconfigPath: "",
+				Error:          fmt.Errorf("unable to list clusters: %w", err),
+			}
+			return
+		}
+		s.Logger.Debug("CAPI: cannot listing v1beta1.Cluster resources, not currently connected to a cluster with CAPI installed")
```

Comment:
It probably would be good to check the error string to if it indicates missing Cluster CRD. Or directly probe the K8s API for the Cluster CRD.

Reply:
great idea, playing around with some ways to do that now, I'll have an update shortly